### DISCUSSION TITLE
Does koboldcpp make use of tensor cores?

### DISCUSSION THREAD

**Discussion by VL4DST3R** [LostRuins/koboldcpp#688] [Q&A]
After playing a bit with "Chat with RTX" that released recently, I was wondering if/when the use of tensor cores will be added to kobold as well? I've found [mentions about this being the case for things like llama.cpp ](https://www.reddit.com/r/LocalLLaMA/comments/1aqh3en/chat_with_rtx_is_very_fast_its_the_only_local_llm/kqcwzfp/) but i've not seen it being mentioned much here except maybe [here](https://github.com/LostRuins/koboldcpp/issues/556).

**Comment by LostRuins** [comment]
It uses them if MMQ is disabled

**Reply by VL4DST3R** [reply]
Oh so using `--usecublas normal` already makes use of them? That's neat! Re-reading the wiki page leads me to understand that it's better to stick to regular cublas then, given how most models are K variants? Also in your testing what does "slightly more VRAM" mean in this case? Something palpable/worth considering?

> mmq is an upstream feature can be added to --usecublas to use quantized matrix multiplication in CUDA during prompt processing, instead of using cuBLAS for matrix multiplication. Experimentally this uses slightly less memory, and is slightly faster for Q4_0 but slower for K-quants. However, for long prompts on new GPUs, cuBLAS is generally faster at the cost of slightly more VRAM (MMQ off).
